### PR TITLE
Improve error handling in get_results method

### PR DESCRIPTION
**Pull request by YNedderhoff**
### tl;dr

This PR enhances the error-handling capabilities of the `get_results` method. 

#### Description

Previously, the method could encounter unhandled exceptions, potentially disrupting the user experience:

```
Traceback (most recent call last):
  File "calibre/gui2/store/search/search.py", line 387, in get_results
  File "calibre/gui2/store/search/models.py", line 97, in add_result
  File "calibre/gui2/store/search/models.py", line 119, in _filter_results
  File "calibre/gui2/store/search/models.py", line 291, in sort
  File "calibre/gui2/store/search/models.py", line 292, in <lambda>
  File "calibre/gui2/store/search/models.py", line 263, in data_as_text
  File "calibre/gui2/store/search/models.py", line 24, in comparable_price
  File "re.py", line 200, in search
TypeError: expected string or bytes-like object
```

#### Screenshot

Screenshot 2023-10-02 at 20 34 24

### Improvement

To address this issue, comprehensive error handling has been added. When processing search results from store plugins, the code now gracefully handles potential TypeError exceptions. In case of an error, an informative error dialog is displayed to the user, providing details about the specific result and the associated store plugin.

This improvement ensures a more reliable and user-friendly experience, enhancing the overall quality of the application.

### CODE REVIEW

**Comment by eli-schwartz** [comment]
I think that spawning an error dialog can be an improvement over the current state of affairs, but would hesitate to call it graceful or "comprehensive error handling" unless it strikes to the heart of the problem.

I'm a lot more curious to know under what circumstances the wrong type shows up -- and which type. Maybe that type should be unwrapped before parsing and then we don't need to raise errors at all?

**Comment by YNedderhoff** [comment]
Fair enough. In my mind this was a quick improvement of the situation for somebody else with more expertise of the application to then build on top on. For instance, this allows for a quick fix manual along the lines of just not using the faulty store, if you know which one it is.
